custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js

```diff
@@ -248,19 +253,22 @@ function ServiceDeliveryDashboardController($rootScope, $scope, $http, $location
 
     function renderPercentageAndPartials(percentage, nominator, denominator, indicator) {
         if (isZeroNullUnassignedOrDataNotEntered(denominator)) {
-            return '<div> No expected ' + indicator + ' </div>';
+            return isMobile ? ('No expected ' + indicator) : '<div> No expected ' + indicator + ' </div>';
         }
         else {
             if (denominator === vm.dataNotEntered) { return vm.dataNotEntered; }
             if (percentage === vm.dataNotEntered) {
                 if (nominator === 0 && denominator === 0) {
-                    return '<div><span>100 %<br>(' + nominator + ' / ' + denominator + ')</span></div>';
+                    return isMobile ? ('100% (' + nominator + ' / ' + denominator + ')') :
+                        '<div><span>100 %<br>(' + nominator + ' / ' + denominator + ')</span></div>';
```

Comment:
could this be extracted to a helper function? does all it do is add <div> and <span> tags to the web version?

Reply:
Sorry, I didn't get what you meant by extracting to helper function. which part?

Reply:
something like:

```javascript
function getDisplay(indicatorHtml) {
  return isMobile ? indicatorHtml : '<div><span>' + indicatorHtml + '</span></div>';
}
```

feel free to do in a follow up if i read it right and it sounds worthwhile. not a big deal, just a lot of repetition in that `indicatorHtml` bit currently